### PR TITLE
Fixing to walk the whole tree when transforming;

### LINKED ISSUE #64: BUG: transform does not walk the tree of all nested tags
Currently the transform expects direct nested tags, and does not allow any other Marko tags.

For example, the `<ebay-select>` expects a direct nested tag `<ebay-select-option>`, and if you add any control tags, like `<if()>` or `<for()>`, you will break rendering.

The fix is to walk the entire tree of children tags and apply the transform to _any_ tag that matches, regardless of whether the nested tag is an immediate child of the tag.

### PR DESCRIPTION
**Pull request by seangates**
## Description
The previous transform was only one level deep, to the immediate children of the given component tag. This transformer now walks the entire tree looking for children that match the pattern of `{parentTag}-` (looking for the hypen, specifically).

Additionally, the tests now cover this nested tag, along with updating the paths for essential Marko APIs used during the transform.

## Context
This was reported from our first user, who couldn't get the nested tags to show the options, even though everything was set up as we would expect.

## References
Closes #64

### CODE REVIEW

File: src/common/transformer/test/test.server.js [hunk 1]
```diff
@@ -11,8 +11,9 @@ try {
     Builder = require('marko/compiler/Builder');
 } catch (e) {
     // v4 paths
-    CompileContext = require('marko/dist/compiler/CompileContext');
-    Builder = require('marko/dist/compiler/Builder');
+    const target = require('marko/env').isDebug ? 'src' : 'dist';
+    CompileContext = require(`marko/${target}/compiler/CompileContext`);
+    Builder = require(`marko/${target}/compiler/Builder`);
```
Comment (DylanPiercey) [inline src/common/transformer/test/test.server.js]: Are we testing that this compiles properly in v3?

Comment (seangates) [reply src/common/transformer/test/test.server.js]: @mlrawlings ☝️ ?

Comment (seangates) [reply src/common/transformer/test/test.server.js]: I am assuming it does compile properly, given that it passes the v3 and v4 integration tests.

**Comment by yomed** [comment]
Did you mean to include the empty example template? 
`3-nested-options/template.marko`

**Comment by seangates** [comment]
@yomed I did not! That should have an example in it. Let me fix that.

**Comment by ianmcburnie** [comment]
Is this meant to go into master?

**Comment by seangates** [comment]
@ianmcburnie Yes, this is a bug that needs to be fixed in the 0.1.x minor version, and therefore would become 0.1.2.

**Pull request merged**